cdap-api/src/main/java/co/cask/cdap/api/security/store/SecureStore.java

```diff
@@ -29,15 +29,22 @@
 public interface SecureStore {
 
   /**
+   * List of all the entries in the secure store.
    * @param namespace The namespace that this key belongs to.
    * @return A list of {@link SecureStoreMetadata} objects representing the data stored in the store.
+   * throws NamespaceNotFoundException If the specified namespace does not exist.
```

Comment:
after discussing, the conclusion is to just throw an Exception and not mention the type, since these exceptions are not in api.